wfe2/wfe.go

```diff
@@ -1070,18 +963,20 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(
 
 	// Handle the revocation request according to how it is authenticated, or if
 	// the authentication type is unknown, error immediately
-	if authType == embeddedKeyID {
-		prob = wfe.revokeCertByKeyID(ctx, jws, request, logEvent)
-		addRequesterHeader(response, logEvent.Requester)
-	} else if authType == embeddedJWK {
-		prob = wfe.revokeCertByJWK(ctx, jws, request, logEvent)
-	} else {
-		prob = probs.Malformed("Malformed JWS, no KeyID or embedded JWK")
+	var err error
+	switch authType {
+	case embeddedKeyID:
+		err = wfe.revokeCertByKeyID(ctx, jws, request, logEvent)
```

Comment:
Yeah, I don't like them either. I'm going to keep the authType names the same because they do reflect the actual structure of the request object (whether it has a `KeyID` field or a `JSONWebKey` field), but I've renamed these functions to better represent when they're used.